### PR TITLE
clang static code analyzer issues

### PR DESCRIPTION
**Pull request by ericwol-msft**
for SDK issue https://github.com/Azure/azure-iot-sdk-c/issues/1298

### CODE REVIEW

File: src/amqp_management.c [hunk 1]
```diff
@@ -227,7 +227,7 @@ static AMQP_VALUE on_message_received(const void* context, MESSAGE_HANDLE messag
                                                     LogError("Could not create status-description amqp value");
                                                     amqp_management->on_amqp_management_error(amqp_management->on_amqp_management_error_context);
                                                     /* Codes_SRS_AMQP_MANAGEMENT_01_135: [ When an error occurs in creating AMQP values (for status code, etc.) `on_message_received` shall call `messaging_delivery_released` and return the created delivery AMQP value. ]*/
-                                                    result = messaging_delivery_released();
+                                                    messaging_delivery_released();
```
Comment (ewertons) [inline src/amqp_management.c:230]: Is this correct? that function is to create an AMQP_VALUE for message released, isn't it?

File: src/amqpvalue.c [hunk 2]
```diff
@@ -6387,11 +6388,6 @@ static int internal_decoder_decode_bytes(INTERNAL_DECODER_DATA* internal_decoder
                                 {
                                     LogError("Could not create inner decoder for list items");
                                     internal_decoder_data->decoder_state = DECODER_STATE_ERROR;
-                                    result = MU_FAILURE;
```
Comment (ewertons) [inline src/amqpvalue.c:6390]: I wonder if valgrind will complain about this.

File: src/amqpvalue.c [hunk 3]
```diff
@@ -6826,11 +6823,6 @@ static int internal_decoder_decode_bytes(INTERNAL_DECODER_DATA* internal_decoder
                                 {
                                     internal_decoder_data->decoder_state = DECODER_STATE_ERROR;
                                     LogError("Could not create inner decoder for array items");
-                                    result = MU_FAILURE;
```
Comment (ewertons) [inline src/amqpvalue.c:6829]: I wonder if valgrind will complain about this.

File: src/amqp_management.c (hunk 1 above)
Comment (ericwol-msft) [reply src/amqp_management.c]: changes to is_error to TRUE

File: src/amqpvalue.c (hunk 2 above)
Comment (ericwol-msft) [reply src/amqpvalue.c:6390]: not an issue

File: src/amqpvalue.c (hunk 3 above)
Comment (ericwol-msft) [reply src/amqpvalue.c:6829]: not an issue

File: src/amqp_definitions.c [hunk 4]
```diff
@@ -146,6 +146,12 @@ ERROR_HANDLE error_create(const char* condition_value)
             }
 
             amqpvalue_destroy(condition_amqp_value);
+
+            if (result != 0)
+            {
+                error_destroy(error_instance);
+                error_instance = NULL;
```
Comment (dcristoloveanu) [inline src/amqp_definitions.c:153]: >error_instance = NULL; [](start = 16, length = 22)

Not needed.

**Comment by dcristoloveanu** [comment]
>             if ((result == 0) && (amqpvalue_set_composite_item(error_instance->composite_value, 0, condition_amqp_value) != 0))

Likely not needed, as result is just set to 0 in ln. 140

---
Refers to: src/amqp_definitions.c:143 in 871aa49. [](commit_id = 871aa49f98c802f969b82d7622303b7a33122891, deletion_comment = False)

**Comment by dcristoloveanu** [comment]
> // This file is generated. DO NOT EDIT it manually.

This file is generated with Text Transform. All changes actually should be done in the template.

---
Refers to: src/amqp_definitions.c:6 in 871aa49. [](commit_id = 871aa49f98c802f969b82d7622303b7a33122891, deletion_comment = False)

File: src/amqpvalue.c [hunk 5]
```diff
@@ -6387,11 +6388,6 @@ static int internal_decoder_decode_bytes(INTERNAL_DECODER_DATA* internal_decoder
                                 {
                                     LogError("Could not create inner decoder for list items");
                                     internal_decoder_data->decoder_state = DECODER_STATE_ERROR;
-                                    result = MU_FAILURE;
-                                }
-                                else
-                                {
-                                    result = 0;
                                 }
```
Comment (dcristoloveanu) [inline src/amqpvalue.c:6391]: Nice fix!

**Comment by dcristoloveanu** [comment]
> // This file is generated. DO NOT EDIT it manually.

I shall Approve based on the assumption that these changes will be done in the template.

---
In reply to: [833098441](https://github.com/Azure/azure-uamqp-c/pull/392#issuecomment-833098441) [](ancestors = 833098441)

---
Refers to: src/amqp_definitions.c:6 in 871aa49. [](commit_id = 871aa49f98c802f969b82d7622303b7a33122891, deletion_comment = False)

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

File: src/amqp_definitions.c (hunk 4 above)
Comment (ericwol-msft) [reply src/amqp_definitions.c:153]: removed

**Pull request merged**